Stream privacy icon in typeaheads

Riken-Shah opened the pull request:
<!-- Describe your pull request here.-->
It adds stream privacy icons in all the places except `Message compose addressee typeahead.` and `Adding streams in SETTINGS / DEFAULT STREAMS.` which are blocked by other PRs.

- [x] Move message typeahead.
-  [x] Move topic (via left sidebar menu) typeahead.
- [x] New stream notifications and New user notifications in SETTINGS / ORGANIZATION SETTINGS. Note that the decorator for the selected stream also needs to be modified to indicate the stream privacy setting, rather than always being shown as a #.
- [x] Stream reference compose box typeahead (i.e. #... typeahead).
- [x]  Top search bar auto-complete




Fixes: <!-- Issue link, or clear description.--> #22355
<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
*Move message*
![Screenshot 2022-09-24 at 1 24 57 AM](https://user-images.githubusercontent.com/63820270/192047048-fbc0b226-1aa0-43ad-9527-042925113732.png)
![Screenshot 2022-10-08 at 3 54 35 PM](https://user-images.githubusercontent.com/63820270/194703120-a67abbed-264f-41cc-a138-b3320ce31c6a.png)


*Move Topic*
![Screenshot 2022-09-24 at 1 25 15 AM](https://user-images.githubusercontent.com/63820270/192047162-9d315323-f53d-4b6f-b540-8f704eed1128.png)
![Screenshot 2022-10-08 at 3 54 43 PM](https://user-images.githubusercontent.com/63820270/194703115-6fe8361d-8caa-406d-8d07-8cf3169aadb2.png)


*Settings ORG*
![settings_org_active_stream](https://user-images.githubusercontent.com/63820270/188557713-8865c1a3-7483-4f3d-92de-e254acdb1cfb.png)
![Screenshot 2022-09-24 at 1 28 53 AM](https://user-images.githubusercontent.com/63820270/192047591-79f70ad2-5d0d-4129-8894-c6b04ed0e70f.png)


*Search*
![Screenshot 2022-09-06 at 9 59 45 PM](https://user-images.githubusercontent.com/63820270/188689011-67412859-1b95-49dd-ab8f-35d3c0fb9993.png)



*Stream reference compose box typeahead*
![Screenshot 2022-09-06 at 9 59 34 PM](https://user-images.githubusercontent.com/63820270/188689092-8daddfc6-1fd5-4560-a8da-c92f71e57e88.png)



**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

N-Shar-ma reviewed (approved):
The code looks good to me!

Riken-Shah commented:
Hey @alya, 
You can give this a review now!

CI fails because of the flake in `admin.ts` (also seen in [here](https://chat.zulip.org/#narrow/stream/43-automated-testing/topic/main.20failing/near/1429780)) it is not related to changes in this PR. 
I am working on a fix for it right now.

timabbott commented:
One thing I notice in the screenshots is the icons aren't horizontally aligned in a way that looks broken. Look at our gear/popover menus for some examples of aligning icons so that the icons are in a column with each other, and the text is too.

Riken-Shah commented:
I have fixed the icon horizontally aligned issue, now both the icons and text are in a column with each other.

I have updated the screenshots. So you take a look again. 

cc: @alya

alya commented:
Nice, I'm super excited for this change! Would it be possible to also make sure we show the stream privacy icon in the selected option in these dropdowns? Specifically, it's currently missing for "Move message" and "Move topic".

Ideally, we'd also show the down-angle on the right side in those dropdowns the way we do in org settings, but that can be a follow-up if needed.

![Screen Shot 2022-09-26 at 11 18 33 PM](https://user-images.githubusercontent.com/2090066/192447901-db0738c3-7fd3-44a2-b275-f3b454c70a99.png)

Riken-Shah commented:
@alya updated the PR with the changes you recommended. PR is ready for another round of review.

alya commented:
Thanks! I have a few comments / questions about the "Move topic" modal:

1. Is it me, or is the globe icon smaller (less tall) than the others?
2. With this change, I think we no longer need the colored bar at the left edge of the stream dropdown. Could we please remove it?
3. Did this PR do anything to change the order of streams in "Move topic"? I don't understand how the streams are ordered there, but it might not be new.

alya commented:
In the top search bar, I think it should be:

Stream **icon** `Stream name`

Not:

**icon** Stream `Stream name`

Riken-Shah commented:
> 1) Is it me, or is the globe icon smaller (less tall) than the others?
> 2) With this change, I think we no longer need the colored bar at the left edge of the stream dropdown. Could we please remove it?

Stream # | Stream Icon
-- | --
<img width="983" alt="Screenshot 2022-10-17 at 1 29 26 AM" src="https://user-images.githubusercontent.com/63820270/196055512-0fe67ec2-4c9b-4144-b409-0c548bffa632.png"> | <img width="983" alt="Screenshot 2022-10-17 at 1 29 32 AM" src="https://user-images.githubusercontent.com/63820270/196055508-fb2fef53-f4a6-433e-9c36-759582f86fb7.png">

How about it now? Done!

> 3) Did this PR do anything to change the order of streams in "Move topic"? I don't understand how the streams are ordered there, but it might not be new.

No

Riken-Shah commented:
> In the top search bar auto-complete, I think it should be:

<img width="983" alt="Screenshot 2022-10-17 at 1 18 45 AM" src="https://user-images.githubusercontent.com/63820270/196056278-5e974c83-68ff-4033-b600-b8754559cfa8.png">

@alya, ready for another round of review!

timabbott commented:
Can we reduce the space between the privacy icons and stream names? I feel like we want to aim for spacing that resembles how we display these in the left sidebar.

alya commented:
Hm, I'm getting an error popping up for some reason when I try to test this PR (pulled down via `tools/fetch-rebase-pull-request 22864`):

![Screen Shot 2022-10-17 at 5 06 10 PM](https://user-images.githubusercontent.com/2090066/196305818-8e832f98-e74b-4900-92f9-b1041aef7297.png)

Riken-Shah commented:
>  Oh, looks like the above error is coming from some bug in main.

Rebased, Should work now.
Please let me if there is anything that needs to be updated.

alya commented:
Thanks! (The bug in `main` is still there, but I can test the PR regardless.)

The one thing I noticed is that the stream symbols in the search dropdown aren't quite aligned right. Different types of symbols are more shifted to the left or right.

![Screen Shot 2022-10-18 at 9 22 09 AM](https://user-images.githubusercontent.com/2090066/196487647-2470dc8d-42cd-40ef-92c5-cb3ba0efb76a.png)

They should all be in one column, like they are in the auto-complete:
![Screen Shot 2022-10-18 at 9 22 20 AM](https://user-images.githubusercontent.com/2090066/196487767-22a43e27-2706-4521-9ffd-07306a62bb76.png)

Riken-Shah commented:
>Both the symbols and the stream names should be in one column, like they are in the auto-complete:


Sorry, forgot to verify that, 

![Screenshot 2022-10-19 at 2 26 37 PM](https://user-images.githubusercontent.com/63820270/196646228-46e715ed-140d-4a06-a36f-45b274ebede5.png)

I've used same styling we are using in left sidebar. 

Ready for another review @alya.

alya commented:
Hm, similar to @timabbott 's comment above, it looks weird to me that there's more space to the right of the symbol than the left. Let's try reducing the right-side space, maybe?

alya commented:
I think it's OK if the spacing doesn't match the left sidebar.

Riken-Shah commented:
> Let's try reducing the right-side space, maybe?

How about now?

Before | After
-- | --
![Screenshot 2022-10-19 at 2 26 37 PM](https://user-images.githubusercontent.com/63820270/196646228-46e715ed-140d-4a06-a36f-45b274ebede5.png) | ![Screenshot 2022-10-19 at 11 33 26 PM](https://user-images.githubusercontent.com/63820270/196770170-d3b9359b-dfe6-4ca5-bb90-5e82b7d60f72.png)

alya commented:
Hm, I think it will look best if there is clearly more space on the left of the icon than on the right, so that it feels connected to the name of the stream.

Riken-Shah commented:
>Hm, I think it will look best if there is clearly more space on the left of the icon than on the right, so that it feels connected to the name of the stream.

Adding more 2px spacing (Currently 4px).

Cool, How about now?

Before | After
-- | --
![Screenshot 2022-10-19 at 2 26 37 PM](https://user-images.githubusercontent.com/63820270/196646228-46e715ed-140d-4a06-a36f-45b274ebede5.png) |  ![Screenshot 2022-10-20 at 7 32 27 PM](https://user-images.githubusercontent.com/63820270/196970357-96ea2842-f66f-42f2-934b-8816fe2721a1.png)

alya commented:
That still looks too centered to me, but @timabbott maybe you can take a look, and perhaps we can tweak the alignment during the review?

Riken-Shah commented:
@timabbott I have rebase this, it's ready for your review.

jychen630 commented:
@Riken-Shah  Just jumping to this PR, do you want to continuet to solve the merge conflict and finish this? Or I can pick it up and finish this :-) I'm an upcoming GSoC applicant.

Riken-Shah commented:
Yeah, it'd be great! I am not sure if I'll get time to work on this soon!
